src/server_manager/cloud/digitalocean_api.ts

```diff
@@ -220,6 +220,8 @@ class RestApiSession implements DigitalOceanSession {
           // this.response may be empty.
           const responseObj = (xhr.response ? JSON.parse(xhr.response) : {});
           resolve(responseObj);
+        } else if (xhr.status === 401) {
+          reject(new XhrError());
         } else {
           // this.response is a JSON object, whose message is an error string.
           const responseJson = JSON.parse(xhr.response);
```

Comment:
This may throw if the response is not JSON. I don't know what you get in the case of authentication error. Perhaps that's the reason you can't catch the authentication errors?